django/cantusdb_project/main_app/tests/test_views.py

```diff
@@ -1966,30 +2019,50 @@ def test_keyword_search_searching_all_fields(self):
         includes_search_term = "brevity is the soul of wit"
         doesnt_include_search_term = "longevity is the soul of wit"
         source = make_fake_source()
+
+        # For all the chants below: incipit is automatically calculated
```

Comment:
I don't understand for this test why we want to set the incipit to be something different than it would be as it is with the signal? It seems like this is testing a situation (where the incipit is created through means other than the signal) that should never happen!

Just looking through these:

For `chant_incipit`, I have a chant whose incipit does include the search term but neither of the full texts do...if this is happening, it's bad!

`chant_ms_spelling` and `chant_std_spelling` could be real objects in the database (the term is somewhere in the either of the full texts but not in the first 5 words), but in that case, why not make the test chants fit that real case rather than what you do here? (e.g. put the search term as word # 6).

Reply:
After looking through the code of the relevant view, I now see what this is testing... but to my mind, the fact that you have to skirt around the function this PR introduces to make this test work points to the fact that the view needs to change (yay using tests to identify these kinds of things!). Don't search `incipit` anymore in this context because it provides no information not in these other fulltext fields.

Reply:
this is a good point. thanks for flagging this - I'll update the view!

Reply:
on second thought, this might be an item for a separate PR.

I haven't started trying to code this, but one thing that occurs to me is that there are several chants, inherited from OldCantus, for which we have only an incipit and no fulltext. So this will require a bit of thought.

(if we end up still searching the incipit field and the fulltext fields, we should at least have a test to check the incipit-yes—fulltext-no case, and not one that tests an impossible situation as we currently have)

Reply:
I see...yes, it seems like it would make sense to keep the view as is for those legacy cases where there is just an incipit, but agree that a change in the test so that the test chant "looks like" that legacy case would make the purpose behind it much clearer.

Reply:
So, I've not touched the corresponding view - we need to search through the incipit field as well as through the fulltext fields - but I've updated the test to clarify the situation in which we might expect to find a match in the `incipit` field but not in the other fields.

Thank you for spotting this!